gen.py

```diff
@@ -210,9 +244,10 @@ def end_enum(self):
             *children, = [x.displayname for x in c.get_children()]
             skip_prefix = len(os.path.commonprefix(children))
 
-            for x in c.get_children():
+            max = sum(1 for _ in c.get_children())
+            for index, x in  enumerate(list(c.get_children())):
```

Comment:
This code currently walks the children twice and then still pulls the full list to memory (`list(..)`). I think we should just collect the list once instead which then allows us to just `len(x)` it. Something like:

```suggestion
            children = list(c.get_children())
            max = len(children)
            for index, x in enumerate(children):
```

Reply:
Looks good; I'll commit to these changes.

Sorry, I don't use Python that much